lib/config_upgrader.js

```diff
@@ -0,0 +1,272 @@
+"use strict";
```

Comment:
As far as I can tell there's no code for config upgrades in the ESLint repo or in any other repo under ESLint org. Unfortunately.

There are a few upgrade guides for each major version of ESLint. They suggest manual config upgrades. This file performs what those guides recommend doing in-memory. This code also gives specific instructions for the end-user what they need to do to their configs to bring them up to the latest version.